### PR TITLE
Migrate Table to DataTable, migrates Pagination as well

### PR DESCRIPTION
**Pull request by jvorcak**
WIP: You can review but a lot of things are subject to changes, so maybe it's better to review a bit later.

- [x] Empty state is not working yet
- [x] There are still about 10 occurences of KowlTable to be migrated

I've tried to use selection mechanism from tanstack-table, but for now we will need to keep the selection logic out of `DataTable` (at least for a first iteration.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -24,11 +24,13 @@
     "dependencies": {
         "@ant-design/icons": "^4.7",
         "@chakra-ui/icons": "^2.1",
+        "@chakra-ui/react": "^2.8.1",
         "@chakra-ui/react-use-disclosure": "^2.1",
         "@heroicons/react": "^1.0.6",
         "@monaco-editor/react": "^4.6",
         "@primer/octicons-react": "^17",
-        "@redpanda-data/ui": "^3.30.1",
+        "@redpanda-data/ui": "file:.yalc/@redpanda-data/ui",
```
Comment (malinskibeniamin) [inline frontend/package.json:32]: let's revert the change to `yalc`, we can use a beta release, or focus on the datatable updates in UI library 1st?

File: frontend/src/components/pages/admin/Admin.Roles.tsx [hunk 2]
```diff
@@ -32,22 +28,19 @@ export class AdminRoles extends Component<{}> {
...
+            size="md"
+            expandRowByClick
+            subComponent={({row: {original: role}}) => <RoleComponent role={role}/>}
             columns={[
-                { width: 1, title: 'Role Name', dataIndex: 'name', sorter: sortField('name') },
-                { title: '', render: _r => (<span></span>) },
+                {
+                    accessorKey: 'name',
+                    header: 'Role Name',
+                    size: Infinity,
```
Comment (malinskibeniamin) [inline frontend/src/components/pages/admin/Admin.Roles.tsx:41]: How does Infinity work here? does it just take all available space?

File: frontend/src/components/pages/admin/Admin.Users.tsx [hunk 3]
```diff
@@ -37,54 +35,47 @@ export class AdminUsers extends Component<{}> {
...
-                        render: (t, r) => {
-                            if (r.internalIdentifier == api.userData?.user.internalIdentifier)
+                        size: 1,
+                        header: 'Identifier',
+                        accessorKey: 'internalIdentifier',
+                        cell: ({row}) => {
+                            if (row.original.internalIdentifier == api.userData?.user.internalIdentifier) {
                                 return (
                                     <span>
+                                        <>
```
Comment (malinskibeniamin) [inline frontend/src/components/pages/admin/Admin.Users.tsx:52]: I think this fragment is not needed

File: frontend/src/components/pages/admin/Admin.Users.tsx [hunk 4]
```diff
@@ -37,54 +35,47 @@ export class AdminUsers extends Component<{}> {
...
-                        defaultIndex={0}
-                        items={
-                            user.grantedRoles.map(r => ({
-                                heading: r.role.name,
-                                description: <RoleComponent role={r.role} grantedBy={r.grantedBy}/>
-                            }))
-                        }
-                    />
-                )}
+                subComponent={({row: {original: user}}) =>   <Accordion
```
Comment (malinskibeniamin) [inline frontend/src/components/pages/admin/Admin.Users.tsx:68]: Can we fix linting here? maybe run `npm run lint` manually if needed

**Comment by malinskibeniamin** [comment]
Tagging @rikimaru0345 and @frankieshakes so we get more eyes on it

File: frontend/src/components/pages/admin/Admin.Roles.tsx (hunk 2 above)
Comment (malinskibeniamin) [reply frontend/src/components/pages/admin/Admin.Roles.tsx:41]: Is thumbs up just a confirmation? :)

File: frontend/src/components/pages/consumers/Modals.tsx [hunk 5]
```diff
@@ -264,36 +263,39 @@ export class EditOffsetsModal extends Component<{
...
-                    description: <Table
-                        size="small"
-                        showSorterTooltip={false}
-                        pagination={{pageSize: 1000, position: ['none', 'none'] as any}}
-                        dataSource={items}
-                        rowKey={r => r.partitionId}
-                        rowClassName={r => (r.newOffset == null ? 'unchanged' : '')}
+                    description: <DataTable<GroupOffset>
+                        size="sm"
+                        showPagination
```
Comment (malinskibeniamin) [inline frontend/src/components/pages/consumers/Modals.tsx:268]: I see previously we had a default page size of 1000, is there any way to provide a default apart from just `showPagination` boolean flag?

File: frontend/src/components/pages/connect/Overview.tsx [hunk 6]
```diff
@@ -226,55 +203,42 @@ class TabTasks extends Component {
...
+                        // Include sorting logic and filter type if necessary
                     },
                     {
-                        title: 'Worker',
-                        dataIndex: 'workerId',
-                        sorter: sortField('workerId'),
-                        filterType: { type: 'enum' }
+                        header: 'Worker',
+                        accessorKey: 'workerId',
+                        // Include sorting logic and filter type if necessary
```
Comment (malinskibeniamin) [inline frontend/src/components/pages/connect/Overview.tsx:235]: Is this a TODO?

File: frontend/src/components/pages/admin/Admin.Roles.tsx (hunk 2 above)
Comment (jvorcak) [reply frontend/src/components/pages/admin/Admin.Roles.tsx:41]: Yes, it takes up all the available space, I'd need some testing with real data to see whether this works as expected in production.

File: frontend/src/components/pages/admin/Admin.Users.tsx (hunk 4 above)
Comment (jvorcak) [reply frontend/src/components/pages/admin/Admin.Users.tsx:68]: I think it's OK now.

File: frontend/src/components/pages/consumers/Modals.tsx (hunk 5 above)
Comment (jvorcak) [reply frontend/src/components/pages/consumers/Modals.tsx:268]: 10 is a default now, Do you suggest I should put back 1000?

File: frontend/package.json (hunk 1 above)
Comment (jvorcak) [reply frontend/package.json:32]: Yeas, please let's just temporarily keep these here until a BETA package is released

File: frontend/src/components/pages/consumers/Modals.tsx (hunk 5 above)
Comment (malinskibeniamin) [reply frontend/src/components/pages/consumers/Modals.tsx:268]: No it's fine, as long as arbitrary default is set, then I think it's fine. @rikimaru0345 ?

File: frontend/src/components/misc/ConfigList.tsx [hunk 7]
```diff
@@ -93,32 +94,89 @@ export function ConfigList({ configEntries, valueDisplay, renderTooltip }: { con
...
+                    label={
+                        <>
+                            <p>Resources can be configured at different levels. Example: A topic config may be inherited from the static broker config.</p>
+                            <p>Valid sources are: Dynamic Topic, Dynamic Broker, Default Broker, Static Broker, Dynamic Broker Logger and Default config.</p>
+                        </>
+                    }
+                    placement="left"
+                    hasArrow
+                >
+                    <InfoCircleFilled style={{color: '#bbbbbb'}}/>
```
Comment (malinskibeniamin) [inline frontend/src/components/misc/ConfigList.tsx:149]: Maybe just use `grey.500`

File: frontend/src/components/misc/ConfigList.tsx [hunk 8]
```diff
@@ -93,32 +94,89 @@ export function ConfigList({ configEntries, valueDisplay, renderTooltip }: { con
...
-                    return false;
+    return (
+        <DataTable<ConfigEntry>
+            data={configEntries}
+            showPagination={false}
+            enableSorting={false}
+            defaultPageSize={10000}
+            size="md"
+            getRowCanExpand={row => (row.original.synonyms?.length ?? 0) > 0 }
+            // getSubRows={row => row.synonyms as ConfigEntry[]}
```
Comment (malinskibeniamin) [inline frontend/src/components/misc/ConfigList.tsx:167]: Is this still required?

File: frontend/src/components/misc/ConfigList.tsx [hunk 9]
```diff
@@ -93,32 +94,89 @@ export function ConfigList({ configEntries, valueDisplay, renderTooltip }: { con
...
+            subComponent={({row}) => {
+                if(!row.original.synonyms?.length) {
+                    return null
                 }
+                return <DataTable<ConfigEntry>
+                    // @ts-ignore TODO - we need to fix types here and find a shared interface
```
Comment (malinskibeniamin) [inline frontend/src/components/misc/ConfigList.tsx:173]: 👍

File: frontend/src/components/pages/admin/Admin.Users.tsx [hunk 10]
```diff
@@ -37,54 +35,45 @@ export class AdminUsers extends Component<{}> {
...
+                        size: 1,
+                        header: 'Identifier',
+                        accessorKey: 'internalIdentifier',
+                        cell: ({row}) => {
+                            if (row.original.internalIdentifier == api.userData?.user.internalIdentifier) {
                                 return (
                                     <span>
                                         <Tooltip label="You are currently logged in as this user" placement="top" hasArrow>
-                                            <UserOutlined style={{ fontSize: '16px', padding: '2px', color: '#ff9e3a' }} />
+                                            <UserOutlined style={{fontSize: '16px', padding: '2px', color: '#ff9e3a'}}/>
```
Comment (malinskibeniamin) [inline frontend/src/components/pages/admin/Admin.Users.tsx:53]: If possible, can we use a matching color from the UI library palette?

File: frontend/src/components/pages/topics/Topic.List.tsx [hunk 11]
```diff
@@ -164,87 +163,58 @@ class TopicList extends PageComponent {
...
+                                            <DeleteDisabledTooltip topic={record}>
+                                                <Button
+                                                    variant="ghost"
+                                                    className="iconButton"
+                                                    onClick={(event) => {
+                                                        event.stopPropagation();
+                                                        this.topicToDelete = record;
+                                                    }}
+                                                >
+                                                    <Icon as={HiOutlineTrash} fontSize="22px"/>
```
Comment (malinskibeniamin) [inline frontend/src/components/pages/topics/Topic.List.tsx:209]: I think we should create an issue because using `fontSize` does work, but ideally we should be using `boxSize`

File: frontend/src/components/misc/ConfigList.tsx (hunk 7 above)
Comment (rikimaru0345) [reply frontend/src/components/misc/ConfigList.tsx:149]: Yes, should use the built in color "names"

File: frontend/src/components/pages/topics/Topic.List.tsx (hunk 11 above)
Comment (rikimaru0345) [reply frontend/src/components/pages/topics/Topic.List.tsx:209]: Ben, just curious: why box size instead of fontsize? I've been using fontsize all the time for icons. What is the difference?

**Comment by jvorcak** [comment]
I've merged this to a branch where we will try to address migration together with other changes.
